### PR TITLE
Add badges to README.md

### PR DESCRIPTION
**Pull request by huyan0**
This PR adds status badges for repo license, build status and code coverage to README.md
This is useful for all developers to get a at-a-glance status of the repository.

### CODE REVIEW

**Comment by reyang** [comment]
The visual [seems inconsistent](https://github.com/huyan0/opentelemetry-cpp/blob/add-badge/README.md), and there is a line break which we want to remove:

image

**Comment by reyang** [comment]
License info is already showing on the home page, we might want to avoid adding duplicated info:

image

File: README.md [hunk 1]
```diff
@@ -1,9 +1,12 @@
 # OpenTelemetry C++
 
-[![Gitter chat][gitter-image]][gitter-url]
+[![Gitter chat](https://badges.gitter.im/open-telemetry/opentelemetry-cpp.svg)](https://gitter.im/open-telemetry/opentelemetry-cpp?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+[![codecov.io](https://codecov.io/gh/open-telemetry/opentelemetry-cpp/branch/master/graphs/badge.svg?)](https://codecov.io/gh/open-telemetry/opentelemetry-cpp/)
+[![Build Status](https://circleci.com/gh/open-telemetry/opentelemetry-cpp.svg?style=shield)](https://circleci.com/gh/open-telemetry/opentelemetry-cpp)
```
Comment (reyang) [inline README.md:5]: Would you change this from CircleCI to GitHub Actions? Thanks!

**Comment by huyan0** [comment]
Thanks for the feedback. I changed the style of the badges and removed the heading "About this project" so that there is one less line break

**Comment by reyang** [comment]
Please rebase and let's wait for a day for other folks to review and comment.

File: README.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # OpenTelemetry C++
 
-[![Gitter chat][gitter-image]][gitter-url]
-
-[gitter-image]: https://badges.gitter.im/open-telemetry/opentelemetry-cpp.svg
-[gitter-url]: https://gitter.im/open-telemetry/opentelemetry-cpp?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge
+[![Gitter chat](https://badges.gitter.im/open-telemetry/opentelemetry-cpp.svg)](https://gitter.im/open-telemetry/opentelemetry-cpp?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+[![codecov.io](https://codecov.io/gh/open-telemetry/opentelemetry-cpp/branch/master/graphs/badge.svg?)](https://codecov.io/gh/open-telemetry/opentelemetry-cpp/)
+[![Build Status](https://action-badges.now.sh/open-telemetry/opentelemetry-cpp)](https://github.com/open-telemetry/opentelemetry-cpp/actions)
+[![Release](https://img.shields.io/github/v/release/open-telemetry/opentelemetry-cpp?include_prereleases&style=)]()
```
Comment (g-easy) [inline README.md:6]: () -> (https://github.com/open-telemetry/opentelemetry-cpp/releases/)

**Pull request merged**